napari/layers/base/base.py

```diff
@@ -905,15 +912,51 @@ def help(self, help_text):
 
     @property
     def interactive(self):
-        """bool: Determine if canvas pan/zoom interactivity is enabled."""
-        return self._interactive
+        warnings.warn(
+            '`Layer.interactive` is deprecated since 0.5.0 and will be removed in 0.6.0.',
+            category=DeprecationWarning,
+        )
+        return self.mouse_pan or self.mouse_zoom
 
     @interactive.setter
     def interactive(self, interactive):
-        if interactive == self._interactive:
+        warnings.warn(
+            '`Layer.interactive` is deprecated since 0.5.0 and will be removed in 0.6.0.',
+            category=DeprecationWarning,
+        )
+        self.mouse_pan = interactive
+        self.mouse_zoom = interactive
+
+    @property
+    def mouse_pan(self):
+        """bool: Determine if canvas interactive panning is enabled with the mouse."""
+        return self._mouse_pan
+
+    @mouse_pan.setter
+    def mouse_pan(self, mouse_pan):
+        if mouse_pan == self._mouse_pan:
             return
-        self._interactive = interactive
-        self.events.interactive(interactive=interactive)
+        self._mouse_pan = mouse_pan
+        self.events.mouse_pan_zoom_toggles(
+            mouse_pan=mouse_pan, mouse_zoom=self.mouse_zoom
+        )
+        self.events.interactive(
+            interactive=self.mouse_pan or self.mouse_zoom
+        )  # Deprecated since 0.5.0
+
+    @property
+    def mouse_zoom(self):
+        """bool: Determine if canvas interactive zooming is enabled with the mouse."""
+        return self._mouse_zoom
+
+    @mouse_zoom.setter
+    def mouse_zoom(self, mouse_zoom):
+        if mouse_zoom == self._mouse_zoom:
+            return
+        self._mouse_zoom = mouse_zoom
+        self.events.mouse_pan_zoom_toggles(
+            mouse_pan=self.mouse_pan, mouse_zoom=mouse_zoom
+        )
```

Comment:
Why `interactive` event is not triggered here?

Reply:
I have added an explanation above. Napari provides the mechanism for suppressing event emission ant this mechanism should be used here.